dispatcher/src/test/java/io/camunda/zeebe/dispatcher/DispatcherTest.java

```diff
@@ -133,6 +133,34 @@ public void shouldNotClaimBeyondPublisherLimit() {
     verify(logBufferPartition0).getTailCounterVolatile();
   }
 
+  @Test
+  public void canClaimFragmentBatch() {
+    // given
+    final int fragmentCount = 2;
+    final int batchLength = dispatcher.getMaxFragmentLength() / 2;
+
+    // when
+    final var canClaimFragmentBatch = dispatcher.canClaimFragmentBatch(fragmentCount, batchLength);
+
+    // then
+    assertThat(canClaimFragmentBatch).isTrue();
+  }
+
+  @Test
+  public void cannotClaimFragmentBatch() {
+    // given - a fragment of max length, unframed
+    final int fragmentCount = 1;
+    final int batchLength = dispatcher.getMaxFragmentLength();
+
+    // when
+    final var canClaimFragmentBatch = dispatcher.canClaimFragmentBatch(fragmentCount, batchLength);
+
+    // then
+    assertThat(canClaimFragmentBatch)
+        .as("cannot claim when the unframed, unaligned batch is the max fragment length")
+        .isFalse();
+  }
```

Comment:
:wrench: We could write a more general test case to verify that we can claim a batch.

The current test case covers a corner case. As a developer, if I don't know the implementation then the test case doesn't make much sense (i.e. why can I claim `2 * max/2` but not `1 * max`?). 

We could leave this test case to cover the corner case. But it may be helpful to write a more obvious test case (e.g. claim `2 * max` or `3 * max/2`).

Reply:
I'm still not happy with the tests, but I agree with your point, it makes sense to distinguish both cases :slightly_smiling_face: